README.md

```diff
@@ -12,61 +13,116 @@ To install run
+    'cachePath' => '/some/directory/openId-cache.php', // Path for caching above discovery document
+    'clientId' => 'client_id', // Client id assigned by authorizer
+    'clientSecret' => 'client_secret', // Client password assigned by authorizer
+ ]);
 
-```
-OPEN_ID_PROVIDER_URL='https://.../.well-known/openid-configuration...'
-OPEN_ID_PROVIDER_CLIENT_ID={app.client.id}
-OPEN_ID_PROVIDER_CLIENT_SECRET={app.client.secret}
-OPEN_ID_PROVIDER_CACHE_PATH='%kernel.cache_dir%/.well_known_cache.php'
+$authUrl = $provider->getAuthorizationUrl();
+
+// direct to $authUrl
 ```
 
 Note that the default response type and mode
 is set in ```OpenIdConfigurationProvider.php```
 
-```
+```sh
 'response_type' => 'id_token',
 'response_mode' => 'query',
 ```
 
-The request should then be handled in some sort of GuardAuthenticator
+### Symfony usage example
+
+In Symfony, we create a login route that when accessed starts
+the authentication flow. Upon creating an instance of
+`OpenIdConfigurationProvider` we configure it with a return URI,
+a cache path and some authorizer details.
+
+```sh
+use ItkDev\OpenIdConnect\Security\OpenIdConfigurationProvider;
+
+/**
+  * @Route("/login", name="login")
+  */
+public function login(SessionInterface $session, array $openIdProviderOptions = []): Response
+{
+    $provider = new OpenIdConfigurationProvider([
+        'redirectUri' => $this->generateUrl('some_route_here', [], UrlGeneratorInterface::ABSOLUTE_URL),
+    ] + $openIdProviderOptions);
+
+    $authUrl = $provider->getAuthorizationUrl();
+    
+    // Set oauth2state and check it upon receiving respose to avoid CSRF
+    $session->set('oauth2state', $provider->getState());
+
+    return new RedirectResponse($authUrl);
+}
+```
+
+The response from the authorizer should then be processed by a
+[Guard Authenticator](https://symfony.com/doc/current/security/guard_authentication.html).
+
+If you wish to see the package used in a Symfony project
+check out [naevnssekretariatet](https://github.com/itk-dev/naevnssekretariatet):
+
+* How to [bind $openIdProviderOptions](https://github.com/itk-dev/naevnssekretariatet/blob/develop/config/services.yaml)
+* How to [set the environment variables](https://github.com/itk-dev/naevnssekretariatet/blob/develop/.env)
+* How to [create provider and redirect user](https://github.com/itk-dev/naevnssekretariatet/blob/develop/src/Controller/DefaultController.php)
+* How to [process and verify response](https://github.com/itk-dev/naevnssekretariatet/blob/develop/src/Security/OpenIdLoginAuthenticator.php)
+
+## Coding standard tests
+
+The following command let you test that the code follows
+the coding standard we decided to adhere to in this project.
+
+* PHP files (PHP-CS-Fixer)
+
+    ```sh
+    ./vendor/bin/php-cs-fixer fix src --dry-run
+    ```
 
-## Tests
+* Markdown files (markdownlint standard rules)
+  
+    ```sh
+    ./node_modules/.bin/markdownlint --fix *.md
```

Comment:
This command could be replaced by `yarn coding-standards-check` as the command is defined in the package.json file.